.github/workflows/tests-pr.yml

```diff
@@ -1,8 +1,9 @@
+name: Tests Pull Requests
+concurrency:
+  group: tests-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
This is what does the work for us. It cancels previous running tests and lints so pushing several times in a row doesn't flood the queue.